plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,176 @@
+package supervisor
+
+import (
+	"net/url"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/kolo/xmlrpc"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server         string   `toml:"url"`
+	PidGather      bool     `toml:"gather_pid"`
+	ExitCodeGather bool     `toml:"gather_exit_code"`
+	UseIdentTag    bool     `toml:"use_identification_tag"`
+	FieldsInc      []string `toml:"fields_includes"`
+	FieldsExc      []string `toml:"fields_excludes"`
+
+	status supervisorInfo
+
+	rpcClient *xmlrpc.Client
+}
+
+type processInfo struct {
+	Name          string `xmlrpc:"name"`
+	Group         string `xmlrpc:"group"`
+	Description   string `xmlrpc:"description"`
+	Start         int32  `xmlrpc:"start"`
+	Stop          int32  `xmlrpc:"stop"`
+	Now           int32  `xmlrpc:"now"`
+	State         int16  `xmlrpc:"state"`
+	Statename     string `xmlrpc:"statename"`
+	StdoutLogfile string `xmlrpc:"stdout_logfile"`
+	StderrLogfile string `xmlrpc:"stderr_logfile"`
+	SpawnErr      string `xmlrpc:"spawnerr"`
+	ExitStatus    int8   `xmlrpc:"exitstatus"`
+	Pid           int32  `xmlrpc:"pid"`
+}
+
+type supervisorInfo struct {
+	StateCode int8   `xmlrpc:"statecode"`
+	StateName string `xmlrpc:"statename"`
+	Ident     string
```

Comment:
Can you please also add an `xmlrpc` annotation for this field?

Reply:
This field not included in supervisor's XML-RPC state struct, so it needs [separate API call](https://github.com/niasar/telegraf/blob/42abc643a18b9c9ea621429ee86e1869c27e499c/plugins/inputs/supervisor/supervisor.go#L93) to get this data.

Reply:
Understood.